### PR TITLE
linear combination of queries

### PR DESCRIPTION
**Pull request by cmacdonald**

### CODE REVIEW

File: pyterrier/rewrite.py [hunk 1]
```diff
@@ -309,3 +309,65 @@ def transform(self, queries_and_docs):
...
+
+    def __init__(self, weights : List[float], format : str = 'terrierql', **kwargs):
+        super().__init__(**kwargs)
+        self.weights = weights
+        self.format = format
+        if format not in ["terrierql", "matchopql"]:
+            raise ValueError("Format must be one of 'terrierql', 'matchopql'")
+
+    def _terrierql(self, row):
+        return "(%s)^%f (%s)^%f" % (
```
Comment (seanmacavaney) [inline pyterrier/rewrite.py:349]: should we be concerned about escaping the query text, is it safe to assume it's already in the target query language?

Comment (cmacdonald) [reply pyterrier/rewrite.py:349]: I think its safe to assume its already in the correct query language.

Comment (seanmacavaney) [reply pyterrier/rewrite.py:349]: alright, then lgtm

**Review by seanmacavaney — APPROVED** [review]

**Comment by cmacdonald** [comment]
Thanks Sean

**Pull request merged**